esp/src/src-react/components/DiskUsage.tsx

```diff
@@ -31,17 +41,78 @@ interface DetailsProps {
 export const Details: React.FunctionComponent<DetailsProps> = ({
     cluster
 }) => {
-    const summary = React.useMemo(() => {
-        const retVal = new DetailsWidget(cluster)
-            .refresh()
-            .on("componentClick", component => {
-                pushUrl(`/machines/${component}/usage`);
+
+    const [data, setData] = React.useState<any[]>([]);
+    const { refreshTable } = useFluentStoreState({});
+
+    //  Grid ---
+    const columns = React.useMemo((): FluentColumns => {
+        return {
+            PercentUsed: {
+                label: nlsHPCC.PercentUsed, width: 50, formatter: (percent) => {
+                    let className = "";
+
+                    if (percent <= 70) { className = "bgFilled bgGreen"; }
+                    else if (percent > 70 && percent < 80) { className = "bgFilled bgOrange"; }
+                    else { className = "bgFilled bgRed"; }
+
+                    return <span className={className}>{percent}</span>;
+                }
+            },
+            Component: { label: nlsHPCC.Component, width: 90 },
+            Type: { label: nlsHPCC.Type, width: 40 },
+            IPAddress: {
+                label: nlsHPCC.IPAddress, width: 140,
+                formatter: (ip) => <Link href={`#/operations/machines/${ip}/usage`}>{ip}</Link>
+            },
+            Path: { label: nlsHPCC.Path, width: 220 },
+            InUse: { label: nlsHPCC.InUse, width: 50 },
+            Total: { label: nlsHPCC.Total, width: 50 },
+        };
+    }, []);
+
+    const refreshData = React.useCallback(() => {
+        machineService.GetTargetClusterUsageEx([cluster])
+            .then(response => {
+                const _data = [];
+                if (response) {
+                    response.forEach(component => {
+                        component.ComponentUsages.forEach(cu => {
+                            const usage = {};
+                            cu.MachineUsages.forEach(mu => {
+                                mu.DiskUsages.forEach(du => {
+                                    usage["PercentUsed"] = Math.round((du.InUse / du.Total) * 100);
+                                    usage["Component"] = cu.Name;
+                                    usage["IPAddress"] = mu.Name;
+                                    usage["Type"] = du.Name;
+                                    usage["Path"] = du.Path;
+                                    usage["InUse"] = Utility.convertedSize(du.InUse);
+                                    usage["Total"] = Utility.convertedSize(du.Total);
+                                    _data.push(structuredClone(usage));
```

Comment:
1.  This will now produce a syntax error:
```
Argument of type '{}' is not assignable to parameter of type 'Row'.
  Property '__hpcc_id' is missing in type '{}' but required in type '{ __hpcc_id: string; }'.
```
2.  structuredClone is a bit "heavy" here, a shallow clone would suffice: `{ ...usage }`
3. Would be better to simply:
```ts
_data.push({
    ...
    PercentUsed: Math.round((du.InUse / du.Total) * 100),
    Component: cu.Name,
    IPAddress: mu.Name,
    ...
});
```